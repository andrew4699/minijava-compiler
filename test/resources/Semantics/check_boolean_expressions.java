class a {

  public static void main(String[] args) {
    System.out.println(1);
  }
}

class b {
  public boolean valid() {
    boolean x; boolean y;
    x = false; y = true;

    if (x && y) {
      System.out.println(1);
    } else {
      System.out.println(0);
    }

    while (x) {
      System.out.println(0);
    }

    // check &&
    x = true && y;
    y = x && false;
    x = true && true;

    // less than is a boolean?
    x = 1 < 2;

    // check !
    y = !!!!!!!!x;

    return true;
  }

  public boolean invalid() {
    boolean x; boolean y;
    int z;
    x = true; y = false; z = 0;

    if (1) {
      // oops
    } else {

    }

    if (z) {
      System.out.println(0);
    } else {
      System.out.println(1);
    }

    while (z) {
      System.out.println(0);
    }

    while (5) {
      System.out.println(0);
    }

    // check &&
    x = 1 && true;
    y = true && 1;
    x = 1 && 1;

    // less than expects int
    x = true < false;
    x = x < y;

    x = !1;

    return false;
  }
}