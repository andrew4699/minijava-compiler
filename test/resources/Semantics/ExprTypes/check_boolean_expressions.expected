[ExprTypesVisitor - 42]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 48]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 54]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 58]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 63]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 64]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 65]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 65]: Expression yielded type int where boolean was expected.
[ExprTypesVisitor - 68]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 69]: Expression yielded type boolean where number was expected.
[ExprTypesVisitor - 71]: Expression yielded type int where boolean was expected.
